be/src/io/cache/block_file_cache.cpp

```diff
@@ -206,7 +202,11 @@ Status BlockFileCache::initialize() {
 Status BlockFileCache::initialize_unlocked(std::lock_guard<std::mutex>& cache_lock) {
     DCHECK(!_is_initialized);
     _is_initialized = true;
-    _storage = std::make_unique<FSFileCacheStorage>();
+    if (is_in_memory()) {
```

Comment:
> check the config::enable_in_memory_file_cache_storage instead of introducing extra functions

the be config will be removed, and use the parse result of cache_base_path instead

> BlockFileCache should not know whether the storage is in-memory or not.
> it is enough that the file cache storage knows what type it is.

It should. Storage is created by BlockFileCache.